includes/gateway/class-omise-payment-installment.php

```diff
@@ -0,0 +1,262 @@
+					'redirect' => $charge['authorize_uri'],
+				);
+			}
+
+			return $this->payment_failed(
+				sprintf(
+					__( 'Please feel free to try submitting your order again, or contact our support team if you have any questions (Your temporary order id is \'%s\')', 'omise' ),
+					$order_id
+				)
+			);
+		}
+
+		/**
+		 * @return void
+		 */
+		public function callback() {
+			if ( ! isset( $_GET['order_id'] ) || ! $order = $this->load_order( $_GET['order_id'] ) ) {
+				wc_add_notice(
+					wp_kses(
+						__( 'We cannot validate your payment result:<br/>Note that your payment might already has been processed. Please contact our support team if you have any questions.', 'omise' ),
+						array( 'br' => array() )
+					),
+					'error'
+				);
+
+				header( 'Location: ' . wc_get_checkout_url() );
+				die();
+			}
+
+			$order->add_order_note( __( 'Omise: Validating the payment result..', 'omise' ) );
+
+			try {
+				$charge = OmiseCharge::retrieve( $order->get_transaction_id() );
+
+				if ( self::STATUS_FAILED == $charge['status'] ) {
+					throw new Exception( $charge['failure_message'] . ' (code: ' . $charge['failure_code'] . ')' );
+				}
+
+				if ( self::STATUS_SUCCESSFUL == $charge['status'] && $charge['paid'] ) {
+					$order->add_order_note(
+						sprintf(
+							wp_kses(
+								__( 'Omise: Payment successful.<br/>An amount of %1$s %2$s has been paid', 'omise' ),
+								array( 'br' => array() )
+							),
+							$order->get_total(),
+							$order->get_order_currency()
+						)
+					);
+
+					$order->payment_complete();
+
+					WC()->cart->empty_cart();
+
+					header( 'Location: ' . $order->get_checkout_order_received_url() );
+					die();
+				}
+
+				if ( self::STATUS_PENDING == $charge['status'] && ! $charge['paid'] ) {
+					$order->add_order_note(
+						wp_kses(
+							__( 'Omise: The payment has been processing.<br/>Due to the installment provider, this might takes a few seconds or an hour. Please do a manual \'Sync Payment Status\' action from the Order Actions panel or check the payment status directly at Omise dashboard again later', 'omise' ),
+							array( 'br' => array() )
+						)
+					);
+
+					$order->update_status( 'on-hold' );
+
+					header( 'Location: ' . $order->get_checkout_order_received_url() );
+					die();
+				}
+
+				throw new Exception( __( 'Note that your payment may have already been processed. Please contact our support team if you have any questions.', 'omise' ) );
+			} catch ( Exception $e ) {
+				wc_add_notice(
+					sprintf(
+						wp_kses(
+							__( 'It seems we\'ve been unable to process your payment properly:<br/>%s', 'omise' ),
+							array( 'br' => array() )
+						),
+						$e->getMessage()
+					),
+					'error'
+				);
+
+				$order->add_order_note(
+					sprintf(
+						wp_kses(
+							__( 'Omise: Payment failed.<br/>%s', 'omise' ),
+							array( 'br' => array() )
+						),
+						$e->getMessage()
+					)
+				);
+
+				$order->update_status( 'failed' );
+
```

Comment:
Is this the normal way to do a redirect in WC? Feels like we should maybe be calling a utility function that does it

Reply:
will check on this one (there is wp_redirect() function but not sure why I chose to implement this way at the first place)

Reply:
<img width="512" alt="Screen Shot 2562-05-13 at 15 51 32" src="https://user-images.githubusercontent.com/2154669/57608391-0ff90400-7597-11e9-8501-9d422d51c296.png">
Juts check WordPress source code. Seems `wp_redirect` is what we should use instead.

Thanks @jonrandy